cmd/debos/debos.go

```diff
@@ -13,12 +13,9 @@ import (
 	"github.com/sjoerdsimons/fakemachine"
 )
 
-func bailOnError(err error, a debos.Action, stage string) {
-	if err == nil {
-		return
-	}
-
-	log.Fatalf("Action `%s` failed at stage %s, error: %s", a, stage, err)
+func bailOnError(context debos.DebosContext, err error, a debos.Action, stage string) {
```

Comment:
This is mostly a nitpick; but the bailOnError is no longer bailing on error, so should really have a different name now.

also wondering if you could make it set the exitcode rather then your error block having to be 3 lines each time, but i don't really see a elegant solution for that unless you pass a reference to exitcode and make it say (completely untested):
```go
if checkError (err, a, stage, *exitcode) {
  return
}
```

I guess it does save a bit of redundant typing.

Reply:
hm... I thought about something like this but it is no so elegant as I wanted to have.
The most elegant solution would be to get rid of functional part from main() to other function. Something like:
```Go
func do_stuff() int {
  defer func(){ /*cleanup*/}()
   // the most of the functional code here
  if err != nil { return 1 }
  return 0
}
func main() {
 os.Exit(do_stuff())
}
```